src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String result = "Report for period " + dateFrom + " - " + dateTo;
+
+        LocalDate localDateFrom = LocalDate.parse(dateFrom,
+                                    DateTimeFormatter.ofPattern("dd.MM.yyyy"));
+        LocalDate localDateTo = LocalDate.parse(dateTo,
+                                    DateTimeFormatter.ofPattern("dd.MM.yyyy"));
+
+        for (String username: names) {
+            int money = 0;
+            
+            for (String jobData: data) {
+                if (jobData.contains(username)) {
+                    String[] parts = jobData.split(" ");
```

Comment:
Remember about informative names of the variables and methods. The variable name 'dd' is not informative. Consider renaming it to something more descriptive, like 'currentDate' or 'workDate'.